compiler/src/iree/compiler/Dialect/Flow/Transforms/ElementwiseOpFusion.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+//===--- ElementwiseOpFusion.cpp --- Pass to fuse elementwise ops --------===//
+//
+// This pass applies the elementwise operation fusion transformation in Linalg
+// with a IREE-custom cost function.
+//
+//===----------------------------------------------------------------------===//
+
+#include "iree/compiler/Dialect/Flow/Transforms/FusionUtils.h"
+#include "iree/compiler/Dialect/Flow/Transforms/Passes.h"
+#include "iree/compiler/Dialect/Flow/Transforms/RegionOpUtils.h"
+#include "llvm/Support/Debug.h"
+#include "mlir/Dialect/Affine/IR/AffineOps.h"
+#include "mlir/Dialect/Linalg/Transforms/Transforms.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+#define DEBUG_TYPE "iree-flow-elementwise-op-fusion"
+
+namespace mlir::iree_compiler::IREE::Flow {
+
+#define GEN_PASS_DEF_ELEMENTWISEOPFUSIONPASS
+#include "iree/compiler/Dialect/Flow/Transforms/Passes.h.inc"
+
+namespace {
+
+class ElementwiseOpFusionPass
+    : public impl::ElementwiseOpFusionPassBase<ElementwiseOpFusionPass> {
+
+public:
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<affine::AffineDialect, linalg::LinalgDialect>();
+  }
```

Comment:
nit: Use the tablegen `dependentDialects` feature to register dialects.

Reply:
ah, I actually did do that. Thanks!